### PR TITLE
Modify the data class

### PR DESCRIPTION
**Pull request by pokk**
as the title.

### CODE REVIEW

File: ext/src/main/java/com/playone/mobile/ext/InvlideValue.kt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 package com.playone.mobile.ext
 
 const val defaultInt = -1
+const val defaultLong = -1L
+const val defaultDouble = -1.0
 const val defaultStr = "invalidString"
```
Comment (lekaha) [inline ext/src/main/java/com/playone/mobile/ext/InvlideValue.kt]: What do you think, use UPPER_SNAKE_CASE for constant variable? 
Ref:
https://android.github.io/kotlin-guides/style.html#constant-names

File: remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt [hunk 2]
```diff
@@ -1,84 +1,42 @@
...
+    var description: String = defaultStr,
+    var date: Long = defaultLong,
+    var updated: Long = defaultLong,
+    var address: String = defaultStr,
+    var longitude: Double = defaultDouble,
+    var latitude: Double = defaultDouble,
+    var limit: Int = defaultInt,
+    var level: Int = defaultInt,
+    var host: String = defaultStr,
+    var userId: String = defaultStr
```
Comment (lekaha) [inline remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt]: Here is my original test code:
```
data class MockDataClass(var _abc: Int = INVALID_INT) {
    var abc by NotValidVar(_abc)
}
```

But maybe I was not writing precisely so made you confused. This is one is better,

```
data class PlayoneModel(
    private var _id: String = defaultStr,
    private var _name: String = defaultStr,
    private var _description: String = defaultStr,
    private var _date: Long = defaultLong,
    private var _updated: Long = defaultLong,
    private var _address: String = defaultStr,
    private var _longitude: Double = defaultDouble,
    private var _latitude: Double = defaultDouble,
    private var _limit: Int = defaultInt,
    private var _level: Int = defaultInt,
    private var _host: String = defaultStr,
    private var _userId: String = defaultStr
) {
    var id by NotValidVar(if (_id != defaultStr) _id else defaultStr)
    var name by NotValidVar(if (_name != defaultStr) _name else defaultStr)
    var description by NotValidVar(if (_description != defaultStr) _description else defaultStr)
    var date by NotValidVar(if (_date != defaultLong) _date else defaultLong)
    var updated by NotValidVar(if (_updated != defaultLong) _updated else defaultLong)
    var address by NotValidVar(if (_address != defaultStr) _address else defaultStr)
    var longitude by NotValidVar(if (_longitude != defaultDouble) _longitude else defaultDouble)
    var latitude by NotValidVar(if (_latitude != defaultDouble) _latitude else defaultDouble)
    var limit by NotValidVar(if (_limit != defaultInt) _limit else defaultInt)
    var level by NotValidVar(if (_level != defaultInt) _level else defaultInt)
    var host by NotValidVar(if (_host != defaultStr) _host else defaultStr)
    var userId by NotValidVar(if (_userId != defaultStr) _userId else defaultStr)
}
```

I guess you would feel it is super troublesome and actually it is using 2x memory size. I will find a better way to solve it.

File: remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt [hunk 3]
```diff
@@ -1,84 +1,42 @@
...
+    var _updated by NotValidVar(updated)
+    var _address by NotValidVar(address)
+    var _longitude by NotValidVar(longitude)
+    var _latitude by NotValidVar(latitude)
+    var _limit by NotValidVar(limit)
+    var _level by NotValidVar(level)
+    var _host by NotValidVar(host)
+    var _userId by NotValidVar(userId)
+
     fun toMap() = hashMapOf("name" to name,
```
Comment (lekaha) [inline remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt]: this is function is necessary?

Comment (pokk) [reply remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt]: Yes, we need this function for updating.

File: ext/src/main/java/com/playone/mobile/ext/InvlideValue.kt [hunk 4]
```diff
@@ -1,4 +1,6 @@
 package com.playone.mobile.ext
 
-const val defaultInt = -1
-const val defaultStr = "invalidString"
\ No newline at end of file
+const val DEFAULT_INT = -1
+const val DEFAULT_LONG = -1L
+const val DEFAULT_DOUBLE = -1.0
+const val DEFAULT_STR = "invalidString"
```
Comment (lekaha) [inline ext/src/main/java/com/playone/mobile/ext/InvlideValue.kt]: I would put some special symbol into, for example `<THIS IS INVALID STRING>`

**Comment by pokk** [comment]
@lekaha 

Modified. Please take a look.

**Pull request merged**